packages/cross_file/test/x_file_html_test.dart

```diff
@@ -61,6 +61,21 @@ void main() {
     test('Stream can be sliced', () async {
       expect(await file.openRead(2, 5).first, equals(bytes.sublist(2, 5)));
     });
+
+    test('Prefers local bytes over path if both are provided', () async {
+      const String text = 'Hello World';
+
+      final XFile file = XFile.fromData(
+        utf8.encode(text),
+        path: 'test/x_file_html_test.dart',
+        name: 'x_file_html_test.dart',
+        length: text.length,
+        mimeType: 'text/plain',
+        lastModified: DateTime.now(),
+      );
+
+      expect(await file.readAsString(), equals(text));
```

Comment:
I'm not sure if we should test the `path` getter here, too?
I think `URL.createObjectURL` returns unique urls, so we can't check equality. At best I can check if the `path` getter is different from the passed in path, when provided?

Reply:
Yes, I agree, the `path` should NOT be what's passed in when creating a blob. Also, you can validate that the path starts by `blob:` IIRC.

Reply:
Done!